lib/shanty_docker_plugin/image_wrapper.rb

```diff
@@ -0,0 +1,108 @@
+require 'docker'
+require 'shanty_docker_plugin/error'
+require 'shanty_docker_plugin/output_parser'
+require 'shanty/logger'
+
+module DockerPlugin
+  # Public: Provides wapper utility around Docker::Image
+  class ImageWrapper
+    include Shanty::Logger
+    include OutputParser
+
+    attr_reader :name, :registry, :tag
+
+    # Public: Initialize the image wrapper
+    #
+    # name      - The name of the image
+    # tag       - The (optional) tag of the image
+    # registry  - The (optional) registry to pull the
+    #             image from
+    def initialize(name, tag = nil, registry = nil)
+      @name = name
+      @registry = registry
+      @tag = tag
+      if registry.nil? && tag.nil?
+        no_registry_or_tag
+      elsif registry.nil?
+        no_registry
+      else
+        registry_and_tag
+      end
+    end
+
+    # Public: Pull the image from the remote registry or dockerhub
+    #
+    # Raises an Error::PullError if the image cannot be found remotely
+    def pull!
+      logger.info("Pulling image #{@image_name}")
+      Docker::Image.create(fromImage: @repo, tag: @tag)
+      @underlying_image = lookup_image
+    rescue Docker::Error::ArgumentError
+      raise(Error::PullError, "Could not pull image #{@long_name}")
+    end
+
+    # Public: Add a tag to the image
+    #
+    # new_tag - New tag String to be set on the image
+    def add_tag!(new_tag)
+      underlying.tag(repo: @repo, tag: new_tag, force: true) unless underlying.nil?
+    end
+
+    def underlying
```

Comment:
I wonder if you should call this `image` instead. It'd be clear what that meant because the class is called `ImageWrapper`.